esp/services/ws_ecl/ws_ecl_service.cpp

```diff
@@ -1568,12 +1568,17 @@ int CWsEclBinding::getGenForm(IEspContext &context, CHttpRequest* request, CHttp
     xform->setParameter("useTextareaForStringArray", "1");
 
 #ifdef _CONTAINERIZED
-    bool isRoxie = wsecl->connMap.getValue(wuinfo.qsetname) != nullptr;
-#else
+    VStringBuffer xpath("queues[@name='%s']", wuinfo.qsetname.get());
+    IPropertyTree *queue = getComponentConfigSP()->queryPropTree(xpath.str());
+    if (queue && strieq(queue->queryProp("@type"), "roxie"))
+        xform->setParameter("includeRoxieOptions", queue->getPropBool("@queriesOnly") ? "2" : "3");
```

Comment:
@afishbeck is this correct.  Does queriesOnly imply that you cannot deploy a query?  Should it return "1" or "3"?

Reply:
Deploying a query isn't the right term here.

In k8s we separated out the roxies that would listen on a workunit queue to run workunits.

So for queriesOnly, you can deploy a query and then send it xml/json/testsocket requests.  But you can't submit and run workunits. 

The opposites is also currently true.  Roxie-workunit components are setup like eclagents and you can't deploy queries to them. 

Roxie itself supports mixed mode so we could easily change that or make it an option (if the option isn't already there) on the roxie side.  We may need a new flag to specify mixed mode in the config so that this code and other places can know the difference though.